lib/services/unified_resource.go

```diff
@@ -213,8 +314,55 @@ func newWatcher(ctx context.Context, resourceCache *UnifiedResourceCache, cfg Re
 	return nil
 }
 
-func resourceKey(resource types.Resource) []byte {
-	return backend.Key(prefix, resource.GetName(), resource.GetKind())
+// resourceName is a unique name to be used as a key in the resources map
+func resourceKey(resource types.Resource) string {
+	return resource.GetName() + "/" + resource.GetKind()
+}
+
+type resourceSortKey struct {
+	ByName []byte
+	ByType []byte
+}
+
+// resourceSortKey will generate a key to be used in the sort trees
+func makeResourceSortKey(resource types.Resource) resourceSortKey {
+	var name, kind string
+	// set the kind to the appropriate "contained" type, rather than
+	// the container type.
+	switch r := resource.(type) {
+	case types.Server:
+		name = r.GetHostname() + "/" + r.GetName()
+		kind = types.KindNode
+	case types.AppServer:
+		app := r.GetApp()
```

Comment:
For this and the `types.DatabaseServer` case, should `app` or `db` ever be nil or would that be invalid?

Reply:
its not valid but some test cases see to throw around `DatabaseServer` and `AppServer` without any contained resources so this is here to help those pass (and I suppose its good to check just in case)